reference/conanfile/tools/google/bazeldeps.rst

```diff
@@ -0,0 +1,332 @@
+    some breaking changes might still happen if necessary to prepare for the *Conan 2.0 release*.
+
+
+Available since: `1.37.0 <https://github.com/conan-io/conan/releases/tag/1.37.0>`_
+
+
+BazelDeps
+---------
+
+The ``BazelDeps`` is the dependencies generator for Bazel. Generates a *<REPOSITORY>/BUILD.bazel* file per dependency,
+where the *<REPOSITORY>/* folder is the Conan recipe reference name by default, e.g., *mypkg/BUILD.bazel*. Apart from
+that, it also generates a *dependencies.bzl* file which contains a Bazel function to load all your Conan dependencies.
+
+The ``BazelDeps`` generator can be used by name in conanfiles:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    class Pkg(ConanFile):
+        generators = "BazelDeps"
+
+.. code-block:: text
+    :caption: conanfile.txt
+
+    [generators]
+    BazelDeps
+
+And it can also be fully instantiated in the conanfile ``generate()`` method:
+
+.. code:: python
+
+    from conan import ConanFile
+    from conan.tools.google import BazelDeps
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+        requires = "zlib/1.2.11"
+
+        def generate(self):
+            bz = BazelDeps(self)
+            bz.generate()
+
+When the ``BazelDeps`` generator is used, every invocation of ``conan install`` will
+generate several bazel files. For the *conanfile.py*
+above, for example:
+
+.. code-block:: bash
+
+    $ conan install .
+    .
+    ├── BUILD.bazel
+    ├── conanfile.py
+    ├── dependencies.bzl
+    └── zlib
+        └── BUILD.bazel
+
+Every :command:`conan install` generates these files:
+
+* *BUILD.bazel*: empty file which is aimed only to be together the *dependencies.bzl* one.
+  More information `here <https://bazel.build/concepts/build-files>`__.
+* *dependencies.bzl*: this file tells your Bazel *WORKSPACE* how to load the dependencies.
+* *zlib/BUILD.bazel*: contains all the targets that you can load from any of your *BUILD* files. More information in
+  :ref:`conan_tools_google_bazeldeps_targets`.
+
+Let's check the content of the files created:
+
+.. code-block:: python
+    :caption: BUILD.bazel
+
+    # This is an empty BUILD file to be able to load the dependencies.bzl one.
+
+.. code-block:: python
+    :caption: dependencies.bzl
+
+    # This Bazel module should be loaded by your WORKSPACE file.
+    # Add these lines to your WORKSPACE one (assuming that you're using the "bazel_layout"):
+    # load("@//conan:dependencies.bzl", "load_conan_dependencies")
+    # load_conan_dependencies()
+
+    def load_conan_dependencies():
+        native.new_local_repository(
+            name="zlib",
+            path="/path/to/conan/package/folder/",
+            build_file="/your/current/working/directory/zlib/BUILD.bazel",
+        )
+
+Given the example above, and imagining that your WORKSPACE is at the same directory, you would have to add these lines in there:
+
```

Comment:
This reads more like a tutorial than a reference for the tool... Maybe we should just enumerate the files that this generator creates, giving a general explanation of what each one of those does and then adding a section (maybe that could be just for the 2.0 docs) explaining how to use this generator or how you have to modify your project to use this generator.

Reply:
Yes, I know, but I thought that Conan 1.x is not going to have that example so better to have more info like that. I agree with you, and there will be an example for Conan 2.x so this is covered, but the problem was for Conan 1.x

Reply:
Yes, probably leave it as-is for Conan 1.X, and improve it moving to the examples in Conan 2.0